spec/models/user_spec.rb

```diff
@@ -47,21 +47,41 @@
     expect(user).not_to be_valid
   end
 
-  it "is valid with only first name, last name, email and password" do
+  it 'is valid with only first name, last name, email and password' do
     user = User.new(first_name: "User", last_name: "Name", email: "user@example.com", password: "password")
     expect(user).to be_valid
   end
 
-  it "correctly validates telephone number" do
-    expect(FactoryBot.build(:user, telephone_number: "+00491766563")).not_to be_valid
-    expect(FactoryBot.build(:user, telephone_number: "0049 177 499 372")).not_to be_valid
-    expect(FactoryBot.build(:user, telephone_number: "0173XX17352")).not_to be_valid
-    expect(FactoryBot.build(:user, telephone_number: "01731557326")).to be_valid
+  describe 'telephone number validation' do
+    it 'should be valid with a fully specified phone number' do
+      expect(FactoryBot.build(:user, telephone_number: "+00491766563")).to be_valid
+    end
+
+    it 'should be valid with a same country phone number' do
+      expect(FactoryBot.build(:user, telephone_number: "01731557326")).to be_valid
+    end
+
+    it 'should not be valid when the number contains spaces' do
+      expect(FactoryBot.build(:user, telephone_number: "0049 177 499 372")).not_to be_valid
```

Comment:
wouldn't it be helpful if we just stripped the spaces somewhere, so that the user does not get errors when entering space-numbers?

Reply:
( https://stackoverflow.com/questions/28621354/strip-phone-number-input-before-validation-in-rails ) maybe?

Reply:
PO says: "It's ok, it's a phone number, and does not have that high a priority"